stdlib/public/core/Map.swift

```diff
@@ -26,11 +26,10 @@ public struct LazyMapGenerator<
   ///   since the copy was made, and no preceding call to `self.next()`
   ///   has returned `nil`.
   public mutating func next() -> Element? {
-    let x = _base.next()
-    if x != nil {
-      return _transform(x!)
+    guard let x = _base.next() else {
+          return nil
     }
-    return nil
```

Comment:
In your example you honestly don't even need the else since `return` in the `if` statement is already exiting the current scope.

```
if let x = _base.next() {
    return _transform(x)
}
return nil
```